Async trait support

Hi,

Are there any plans to support async trait like below? I wonder it's easily added or not.

```
#![feature(async_fn_in_trait)]

use ambassador::delegatable_trait;
use ambassador::Delegate;

pub struct Base {}

#[delegatable_trait]
pub trait Hello {
    async fn hello(&self);
}

impl Hello for Base {
    async fn hello(&self) {}
}

#[derive(Delegate)]
#[delegate(Hello)]
pub struct Dram {
    inner: Base,
}

fn main() {
    let d = Dram { inner: Base {} };

    d.hello();
}
```

Currently it failed with

```
error[E0308]: mismatched types
  --> src/main.rs:8:1
   |
8  | #[delegatable_trait]
   | ^^^^^^^^^^^^^^^^^^^^ expected `()`, found opaque type
...
17 | #[derive(Delegate)]
   |          -------- in this derive macro expansion
   |
note: while checking the return type of the `async fn`
  --> src/main.rs:14:27
   |
14 |     async fn hello(&self) {}
   |                           ^ checked the `Output` of this `async fn`, found opaque type
   = note: expected unit type `()`
            found opaque type `impl Future<Output = ()>`
   = note: this error originates in the macro `ambassador_impl_Hello` which comes from the expansion of the derive macro `Delegate` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider `await`ing on the `Future`
   |
8  | #[delegatable_trait].await
   |                     ++++++
help: consider using a semicolon here
   |
8  | #[delegatable_trait];
   |                     +
```

Thanks

--- a/ambassador/src/register.rs
+++ b/ambassador/src/register.rs
@@ -321,7 +321,12 @@ fn build_method_invocation(
 
     let generics = method_sig.generics.split_for_impl().1;
     let turbofish = generics.as_turbofish();
+    let post = if original_method.sig.asyncness.is_some() {
+        quote!(.await)
+    } else {
+        quote!()
+    };
 
-    let method_invocation = quote! { #field_ident.#method_ident #turbofish(#argument_list) };
+    let method_invocation = quote! { #field_ident.#method_ident #turbofish(#argument_list) #post };
     method_invocation
 }

--- a/ambassador/tests/compiletest.rs
+++ b/ambassador/tests/compiletest.rs
@@ -6,7 +6,7 @@ fn run_mode(mode: &'static str) {
     let mut config = compiletest::Config::default();
 
     config.mode = mode.parse().expect("Invalid mode");
-    config.target_rustcflags = Some("-L ../target/debug/deps".to_owned());
+    config.target_rustcflags = Some("-L ../target/debug/deps --edition 2018".to_owned());
     config.src_base = PathBuf::from(format!("tests/{}", mode));
     config.link_deps(); // Populate config.target_rustcflags with dependencies on the path
 

--- a/ambassador/tests/run-pass/async.rs
+++ b/ambassador/tests/run-pass/async.rs
@@ -1,0 +1,51 @@
+extern crate ambassador;
+
+use ambassador::*;
+
+pub struct Base1 {}
+
+#[derive(Delegate)]
+#[delegate(Hello, target = "self")]
+pub struct Base2 {}
+
+impl Base2 {
+    async fn hello(&self) {}
+}
+
+#[delegatable_trait]
+trait Hello {
+    async fn hello(&self);
+}
+
+impl Hello for Base1 {
+    async fn hello(&self) {}
+}
+
+#[derive(Delegate)]
+#[delegate(Hello)]
+pub struct Dram {
+    inner: Base1,
+}
+
+#[derive(Delegate)]
+#[delegate(Hello)]
+pub enum Either {
+    Base1(Base1),
+    Base2(Base2),
+}
+
+struct OpBase(Option<Base1>);
+
+#[delegate_to_methods]
+#[delegate(Hello, target_ref = "unwrap_ref")]
+impl OpBase {
+    fn unwrap_ref(&self) -> &Base1 {
+        self.0.as_ref().unwrap()
+    }
+}
+
+fn main() {
+    let d = Dram { inner: Base1 {} };
+
+    let _ = d.hello();
+}

